src/net.rs

```diff
@@ -1761,4 +1765,77 @@ pub(crate) mod test {
         router2.shutdown().await.e()?;
         Ok(())
     }
+
+    #[tokio::test]
+    #[traced_test]
+    async fn gossip_rely_on_gossip_discovery() -> n0_snafu::Result<()> {
+        let rng = &mut rand_chacha::ChaCha12Rng::seed_from_u64(1);
+
+        async fn spawn(
+            rng: &mut impl CryptoRng,
+        ) -> n0_snafu::Result<(NodeId, Router, Gossip, GossipSender, GossipReceiver)> {
+            let topic_id = TopicId::from([0u8; 32]);
+            let ep = Endpoint::builder()
```

Comment:
`empty_builder`?

Reply:
This PR is still on the last released iroh, will fix in #110